### PR TITLE
Add `env variables` management via maya installer

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit has the following feature:
- Makes use of maya installer to set environment variables that are not already set
- The intention is to simplify learning curve associated with environment variables via Maya Installer
- This also reduces the exposure of various environment variables to the users that can lead to un-necessary confusions leading to misconfigurations and hence errors
- Makes necessary movement of existing environment variables

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/maya-apiserver/spc-watcher/lease.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 	"fmt"
 	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
 	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
-	env "github.com/openebs/maya/pkg/env/v1alpha1"
+	menv "github.com/openebs/maya/pkg/env/v1alpha1"
```
Comment (sonasingh46) [inline cmd/maya-apiserver/spc-watcher/lease.go]: why this change ?

File: pkg/install/v1alpha1/version.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const (
+	version070     version = "0.7.0"
+	invalidVersion version = "invalid.version"
+)
+
+func CurrentVersion() version {
+	return version(mver.GetVersion())
+}
+
+func Version(version string) version {
```
Comment (kmova) [inline pkg/install/v1alpha1/version.go]: @AmitKumarDas - Can we add an unit test that fails if an invalid version is returned? 

When a version file is changed and if the entry for the version is not updated in this file, we could get into a situation where CurrentVersion returns invalid. Adding this UT will make sure that travis will fail, if only version file is changed and not this code.

File: pkg/client/k8s/v1alpha1/config.go [hunk 3]
```diff
@@ -34,8 +34,8 @@ type ConfigGetter func() (*rest.Config, error)
 // environment values
 func WithEnvConfigGetter() ConfigGetter {
 	return func() (*rest.Config, error) {
-		k8sMaster := env.Get(string(K8sMasterIPEnvironmentKey))
-		kubeConfig := env.Get(string(KubeConfigEnvironmentKey))
+		k8sMaster := env.Get(K8sMasterIPEnvironmentKey)
```
Comment (sonasingh46) [inline pkg/client/k8s/v1alpha1/config.go]: Why `menv`  not used here ? Above , we are using `menv.get`

File: pkg/env/v1alpha1/env.go [hunk 4]
```diff
@@ -18,43 +18,141 @@ package v1alpha1
...
+}
+
+// Truthy returns boolean based on the environment variable's value
+//
+//  The value can be truthy or falsy
+func Truthy(envKey ENVKey) (truth bool) {
+	v, found := Lookup(envKey)
+	if !found {
+		return
+	}
```
[the remark this answers is not in the document]
Comment (princerachit) [reply pkg/env/v1alpha1/env.go]: Default value is false.

File: pkg/env/v1alpha1/env.go [hunk 5]
```diff
@@ -18,43 +18,141 @@ package v1alpha1
...
+	truth, _ = strconv.ParseBool(v)
+	return
+}
+
+// LookupS looks up an environment variable and returns false if environment
+// variable is not present
+func LookupS(envKey ENVKey) string {
```
Comment (princerachit) [inline pkg/env/v1alpha1/env.go]: Some better name required. Suffix 'S' not making any sense.

File: pkg/client/k8s/v1alpha1/unstructured.go [hunk 6]
```diff
@@ -165,6 +165,15 @@ func IsNamespaceScoped(given *unstructured.Unstructured) bool {
 	return kind(given.GetKind()).isNamespaced()
 }
 
+// IsCASTemplate flags if the given unstructured instance is a CAS Template
+// scoped
```
Comment (sonasingh46) [inline pkg/client/k8s/v1alpha1/unstructured.go]: I think the comment should be `...CAS Template Kind` if we are trying to convey this.

File: pkg/install/v1alpha1/cstor_sparse_pool_claim_0.7.0_test.go [hunk 7]
```diff
@@ -20,40 +20,113 @@ import (
...
-func TestIsCstorSparsePoolEnabled(t *testing.T) {
-	// Set env variable to enable sparse pool creation
-	setEnv()
-	result := IsCstorSparsePoolEnabled()
-	if result != true {
-		t.Errorf("Test failed as the env variable for cstor sparse pool creation is true but function returned false")
+func TestIsCstorSparsePool(t *testing.T) {
+	tests := map[string]struct {
+		value     string
+		isenabled bool
```
Comment (princerachit) [inline pkg/install/v1alpha1/cstor_sparse_pool_claim_0.7.0_test.go]: isEnabled

File: pkg/env/v1alpha1/env.go [hunk 8]
```diff
@@ -18,43 +18,141 @@ package v1alpha1
...
-	// openebs serviceaccount
-	EnvKeyForOpenEBSServiceAccount ENVKey = "OPENEBS_SERVICE_ACCOUNT"
+	// OpenEBSNamespace is the environment variable to get openebs namespace
+	//
+	// This environment variable is set via kubernetes downward API
+	OpenEBSNamespace ENVKey = "OPENEBS_NAMESPACE"
+
+	// OpenEBSMayaPodName is the environment variable to get maya-apiserver pod
+	// name
+	OpenEBSMayaPodName ENVKey = "OPENEBS_MAYA_POD_NAME"
```
Comment (prateekpandey14) [inline pkg/env/v1alpha1/env.go]: add ---- set via Maya apiserver pod downward API ?

File: pkg/env/v1alpha1/env.go [hunk 9]
```diff
@@ -18,43 +18,141 @@ package v1alpha1
...
+func Set(envKey ENVKey, value string) (err error) {
+	return os.Setenv(string(envKey), value)
+}
 
 // Get fetches value from the provided environment variable
 //
 // NOTE:
 //  This is an implementation of EnvironmentGetter
-func Get(envKey string) (value string) {
-	return getEnv(envKey)
```
Comment (prateekpandey14) [inline pkg/env/v1alpha1/env.go]: Again this has to be generic IMO ....b/c if someone wants to use this helper functions , the key has to be of type ENVKey.

File: pkg/env/v1alpha1/env.go (hunk 8 above)
Comment (AmitKumarDas) [reply pkg/env/v1alpha1/env.go]: Thanks

File: pkg/env/v1alpha1/env.go (hunk 5 above)
Comment (AmitKumarDas) [reply pkg/env/v1alpha1/env.go]: Will do something about it @princerachit

File: pkg/install/v1alpha1/cstor_sparse_pool_claim_0.7.0_test.go [hunk 10]
```diff
@@ -20,40 +20,113 @@ import (
...
 func TestCstorSparsePoolSpc070(t *testing.T) {
-	// Set Env variable to enable sparse pool creation
-	setEnv()
-	listItems := CstorSparsePoolSpc070()
-	if len(listItems.Items) == 0 {
-		t.Errorf("Expected non empty string list")
+	tests := map[string]struct {
+		value    string
+		expected int
+		iserr    bool
```
Comment (prateekpandey14) [inline pkg/install/v1alpha1/cstor_sparse_pool_claim_0.7.0_test.go]: nit: isErr ?

File: pkg/env/v1alpha1/env.go [hunk 11]
```diff
@@ -18,43 +18,141 @@ package v1alpha1
...
+func Get(envKey ENVKey) (value string) {
+	return getEnv(string(envKey))
```
Comment (prateekpandey14) [inline pkg/env/v1alpha1/env.go]: Again this has to be generic IMO ....b/c if someone wants to use this helper functions , the key has to be of type ENVKey.

File: pkg/install/v1alpha1/env_registrar.go [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+	l.Items = append(l.Items, &env{Key: InstallerConfigName, Value: "maya-install-config-default-0.7.0"})
+	return
+}
+
+// envList070 returns a list of environment variable info specific to 0.7.0
+// version
+func envList070() (l *envList) {
+	l = &envList{}
+	l.Items = append(l.Items, &env{Key: DefaultCstorSparsePool, Value: "false"})
+	l.Items = append(l.Items, &env{Key: menv.CASTemplateFeatureGateENVK, Value: "true"})
```
Comment (prateekpandey14) [inline pkg/install/v1alpha1/env_registrar.go]: Feature Gate is needed to set here ....?

File: pkg/env/v1alpha1/env.go (hunk 11 above)
Comment (AmitKumarDas) [reply pkg/env/v1alpha1/env.go]: It needs to be generic to this project to begin with. Let this project make use of this type to make use of the exposed os env functions.

File: pkg/install/v1alpha1/env_registrar.go (hunk 12 above)
Comment (AmitKumarDas) [reply pkg/install/v1alpha1/env_registrar.go]: It will be removed later. Clean Up will be done.
